### PR TITLE
Fix potential bugs

### PR DESCRIPTION
**Pull request by blindpirate**
This is fixing some potential bugs from https://sonarcloud.io/dashboard?id=danielsun1106_groovy

@danielsun1106

### CODE REVIEW

File: src/main/java/groovy/lang/ExpandoMetaClass.java [hunk 1]
```diff
@@ -477,11 +477,11 @@ public static void disableGlobally() {
...
-    /* (non-Javadoc)
-     * @see groovy.lang.MetaClassImpl#initialize()
+    /**
+     * {@inheritDoc}
      */
-
-    public void initialize() {
+    @Override
+    public synchronized void initialize() {
         try {
```
Comment (danielsun1106) [inline src/main/java/groovy/lang/ExpandoMetaClass.java]: The method implements synchronization via ` lock` API,  the `synchronized` seems redundant.

Comment (blindpirate) [reply src/main/java/groovy/lang/ExpandoMetaClass.java]: You're right. Removed `synchronized.`

File: src/main/java/groovy/lang/ObjectRange.java [hunk 2]
```diff
@@ -302,7 +302,7 @@ public int size() {
                 // let's fast calculate the size
                 final BigDecimal fromNum = new BigDecimal(from.toString());
                 final BigDecimal toNum = new BigDecimal(to.toString());
-                final BigInteger sizeNum = toNum.subtract(fromNum).add(new BigDecimal(1.0)).toBigInteger();
+                final BigInteger sizeNum = toNum.subtract(fromNum).add(BigDecimal.valueOf(1.0)).toBigInteger();
```
Comment (paulk-asert) [inline src/main/java/groovy/lang/ObjectRange.java]: We can probably just use: BigDecimal.ONE these days.

File: src/main/java/groovy/lang/ExpandoMetaClass.java [hunk 3]
```diff
@@ -477,10 +477,10 @@ public static void disableGlobally() {
         ExpandoMetaClassCreationHandle.disable();
     }
 
-    /* (non-Javadoc)
-     * @see groovy.lang.MetaClassImpl#initialize()
+    /**
+     * {@inheritDoc}
```
Comment (paulk-asert) [inline src/main/java/groovy/lang/ExpandoMetaClass.java]: This is better for our "api" Javadoc but we currently don't support inheritDoc within Groovydoc:
https://issues.apache.org/jira/browse/GROOVY-3782
so it leaves a bit of noise in the "gapi" documentation - still, not a huge issue I guess.

File: src/main/java/groovy/lang/ObjectRange.java (hunk 2 above)
Comment (blindpirate) [reply src/main/java/groovy/lang/ObjectRange.java]: Nice catch! Will change to `BigDecimal.ONE`.

File: src/main/java/groovy/lang/ExpandoMetaClass.java (hunk 3 above)
Comment (blindpirate) [reply src/main/java/groovy/lang/ExpandoMetaClass.java]: But this is a java file... do you mean javadocs in java files are also processed by `groovydoc`?

Comment (paulk-asert) [reply src/main/java/groovy/lang/ExpandoMetaClass.java]: Yes, the gapi (Groovydoc ) is currently the combined documentation produced by converting java -> groovy and then running groovydoc on all. But as I said, not a big deal for this PR. It was more to inform why it was the way it was.

Comment (blindpirate) [reply src/main/java/groovy/lang/ExpandoMetaClass.java]: Thanks for the clarification.

**Comment by danielsun1106** [comment]
Merged. Thanks.

**Pull request merged**